### PR TITLE
fix: Make oAuth paths more consistent and improve authentication client

### PR DESCRIPTION
**Pull request by daffl**
This pull request makes the oAuth callback paths consistent. It uses the redirects  discussed in https://github.com/simov/grant/pull/116 to make things more straightforward and backwards compatible. Also changes the base path to `/oauth` as previously discussed with @KidkArolis (it can be made backwards compatible by setting `authentication.oauth.defaults.path` to `/auth`).

The relevant oAuth paths now are:

- `http(s)://<host>/oauth/<provider>` to initiate the oAuth flow
- `http(s)://<host>/oauth/<provider>/callback` as the callback path that should be set in the oAuth application settings
- `http(s)://<host>/oauth/<provider>/authenticate` as the internal redirect

### CODE REVIEW

**Comment by KidkArolis** [comment]
Looks good. Are you gonna be adding backwards compatible redirects?

**Comment by daffl** [comment]
I tried but couldn't get it to work. oAuth always errors when I try to redirect from a backwards compatible callback path. At the moment it looks like this is only possible if my PR in Grant gets accepted. I might submit it again to discuss more.

**Comment by daffl** [comment]
@KidkArolis Allright, got some help and managed to make the paths backwards compatible (updated in the description).

File: packages/authentication-oauth/test/express.test.ts [hunk 1]
```diff
@@ -14,35 +14,35 @@ describe('@feathersjs/authentication-oauth/express', () => {
...
       assert.ok(/Cannot GET/.test(error.response.data));
       delete app.get('authentication').oauth.redirect;
     }
   });
 
-  it('auth/test/authenticate with error', async () => {
+  it('oauth/test/authenticate with error', async () => {
     try {
-      await axios.get('http://localhost:9876/oauth/connect/test/authenticate');
+      await axios.get('http://localhost:9876/oauth//test/authenticate');
```
Comment (KidkArolis) [inline packages/authentication-oauth/test/express.test.ts]: Why is there a double slash `//` here?

Comment (daffl) [reply packages/authentication-oauth/test/express.test.ts]: Definitely a find and replace issue 😄

Comment (KidkArolis) [reply packages/authentication-oauth/test/express.test.ts]: Ah, I thought this was an assertion, which was a bit more worrying, but it's just a get. In any case..

**Pull request merged**